### ISSUE TITLE
IndexedDB Recognition fails for Safari 11+

### ISSUE THREAD

**Issue by fifafu** [localForage/localForage#766]
Tested on macOS with Safari 11.0.1 and iOS 11.2

Safari itself is detected correctly, however the hasFetch check doesn't work.

```var hasFetch = typeof fetch === 'function' && fetch.toString().indexOf('[native code') !== -1;```
The first part evaluates to true, but the [native code] can not be found and thus hasFetch is false.

**Comment by thgreasi** [comment]
Can you share the result of `fetch.toString()`?

**Comment by fifafu** [comment]
Sorry should have done that immediately.
It turns out that Angular 5 (or something Angular uses) is monkey patching fetch to intercept it, and thus breaks localforage.

You can test it e.g. on https://angular.io:

```
> console.log(fetch.toString())
[Log] function (){var t=e.apply(this,arguments);if(t instanceof E)return t;var n=t.constructor;return n[Z]||p(n),t}
```

It works as expected on non Angular apps

**Comment by fifafu** [comment]
Actually the patching is done by Zone.js (which is used by Angular).
See https://github.com/angular/zone.js/blob/326a07fb9095e4a87ff7561cb45fe1d4917c2174/lib/common/promise.ts

**Comment by thgreasi** [comment]
I guess we can plan to change that in v2.0 to use a regex and explicitelly detect safari < 10.2.
We should probably also release a new version for the  localforage compatibility plugin.
@tofumatt what do you think?

**Comment by tofumatt** [comment]
I agree. Our feature detection is roundabout (we aren’t actually checking for fetch because we *use* it) and it’s probably better to do a new compatibility release too. 👍🏻

**Comment by lfairy** [comment]
Is the IndexedDB support still buggy on iOS Safari 10.0 through 10.2, or do we just not have an easy way of detecting them?

**Comment by thgreasi** [comment]
It's still light in safari <10.2 and we didn't expect monkey patching the fetch API to be so widely used and stop our 10.2 detection from working.

**Comment by thgreasi** [comment]
We could do something like https://github.com/DamonOehlman/detect-browser/blob/master/index.js for our safari detection.

**Comment by thgreasi** [comment]
See: [Pouch DB merged PR](https://github.com/pouchdb/pouchdb/pull/7086)
See: [Pouch DB dropping Safari < 10.1 support and the `fetch` check](https://github.com/pouchdb/pouchdb/pull/7086)

**Comment by thgreasi** [comment]
See: [Pouch DB merged PR](https://github.com/pouchdb/pouchdb/pull/6938)
See: [Pouch DB dropping Safari < 10.1 support and the `fetch` check](https://github.com/pouchdb/pouchdb/pull/7086)

**Comment by thgreasi** [comment]
Interestingly enough [canIUse](https://caniuse.com/#feat=indexeddb) shows Safari v10.0 as fully compatible as well, but PouchDB guys test cases only passed on 10.1+ .

**Comment by thgreasi** [comment]
@fifafu, @lfairy and anybody interested on this issue getting resolved, can you try out #799 and share some feedback?
